app/js/sagas/orderSaga.js

```diff
@@ -0,0 +1,82 @@
+import {
+  call,
+  take,
+  takeEvery,
+  fork,
+  put,
+  cancel,
+  select,
+} from 'redux-saga/effects';
+import { format, addMinutes } from 'date-fns';
+import encounterRest from '../rest/encounterRest';
+import { getDateRange } from '../utils/helpers';
+import { DISCONTINUE_ORDER } from '../actions/actionTypes';
+import fetchLabOrders from '../actions/labOrders/fetchLabOrders';
+
+
+// eslint-disable-next-line consistent-return
+function* getMatchingEncounter(order) {
+  const state = yield select();
+  const patientUuid = order.patient;
+  const encounterTypeUuid = order.encounterType;
+  const locationUuid = order.location.uuid;
+  const provider = state.encounterRoleReducer.encounterRole.uuid;
+  const duration = format(addMinutes(new Date(), -60));
+
+  try {
+    const encounterResponse = yield call(
+      encounterRest.fetchEncountersByPatient,
+      patientUuid,
+      encounterTypeUuid,
+      provider,
+      locationUuid,
+    );
+    if (encounterResponse.results.length) {
+      const matched = getDateRange(
+        encounterResponse.results,
+        duration,
+        new Date(),
+        'encounterDatetime',
+      );
+
+      if (matched.length) {
```

Comment:
I'd add a comment here like "if we find multiple matching encounters, just return first one"  (which I think is fine, but good to note that we realize we are just taking an arbitrary encounter in case we want to refine this logic in the future.